### PR TITLE
[ feature ]: Added Tilling problem ( domino tiling ) in backtracking algorithm folder

### LINKED ISSUE #1884: [Feature]: Want to add Tiling problem ( Domino tiling ) doc. in Backtracking-algorithm folder
### Feature Name

Tiling problem ( domino tiling )

### Feature Description

The Domino Tiling problem involves filling a 2 x n grid using 1 x 2 dominoes, which can be placed horizontally or vertically. The challenge is to calculate the number of ways to completely tile the grid.

### Motivation

Adding this problem to Algo helps users practice dynamic programming and recursion techniques while solving a common problem in grid-based puzzles. It improves problem-solving skills and has real-world applications in space optimization and puzzle games.

### Implementation Suggestions (Optional)

Use a dynamic programming approach to solve the problem efficiently. Store the number of ways to tile smaller grids in a table, iterating to solve for larger grids.

Time Complexity: O(n)
Space Complexity: O(n) (can be optimized)

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [X] Other

### References (Optional)

https://www.geeksforgeeks.org/tiling-problem/

### PR DESCRIPTION
**Pull request by Jivan052**
### Description: 
I have added Tiling-problem doc. in backtracking folder.

The Domino Tiling Problem has been added to Algo, providing a dynamic programming solution to calculate the number of ways to tile a 2 x n grid using 1 x 2 dominoes. This feature helps users practice dynamic programming and recursion techniques, with real-world applications in space optimization and puzzle games. By introducing this classic problem, we aim to enhance problem-solving skills and provide a practical challenge for algorithm enthusiasts.

Fixes #1884

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/backtracking algorithms/Tiling-problem.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+
+id: Tiling-Problem-Domino-Tiling
```
Comment (ajay-dhangar) [inline docs/backtracking algorithms/Tiling-problem.md]: replace `id: Tiling-Problem-Domino-Tiling` with `id:  tiling-problem-domino-tiling`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/backtracking algorithms/Tiling-problem.md (hunk 1 above)
Comment (Jivan052) [reply docs/backtracking algorithms/Tiling-problem.md]: changes has been made. you can review it.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**